tests/test_soft_bfd.py

```diff
@@ -0,0 +1,189 @@
+        self.remove_bfd_session(bfd_session_key)
+        self.sdb.wait_for_deleted_entry(SOFT_BFD_STATE_TABLE, session)
+
+    def test_addRemoveBfdSession_interface(self, dvs):
+        self.setup_db(dvs)
+        bfd_session_key = "default:Ethernet0:10.0.0.2"
+        bfdSessions = self.get_exist_bfd_session()
+
+        # Create BFD session
+        fieldValues = {"local_addr": "10.0.0.1", "dst_mac": "00:02:03:04:05:06", "type": "async_passive"}
+        self.create_bfd_session("default:Ethernet0:10.0.0.2", fieldValues)
+        self.sdb.wait_for_n_keys(SOFT_BFD_STATE_TABLE, len(bfdSessions) + 1)
+
+        # Check created BFD session in STATE_DB
+        createdSessions = self.get_exist_bfd_session() - bfdSessions
+        assert len(createdSessions) == 1
+
+        session = createdSessions.pop()
+
+        # Check STATE_DB entry related to the BFD session
+        self.check_state_bfd_session_value(bfd_session_key, fieldValues)
+
+        # Remove the BFD session
+        self.remove_bfd_session(bfd_session_key)
+        self.sdb.wait_for_deleted_entry(SOFT_BFD_STATE_TABLE, session)
+
+    def test_multipleBfdSessions(self, dvs):
+        self.setup_db(dvs)
+        bfdSessions = self.get_exist_bfd_session()
+
+        # Create BFD session 1
+        key1 = "default:default:10.0.0.2"
+        fieldValues = {"local_addr": "10.0.0.1"}
+        self.create_bfd_session(key1, fieldValues)
+        self.sdb.wait_for_n_keys(SOFT_BFD_STATE_TABLE, len(bfdSessions) + 1)
+
+        # Checked BFD session 1 in STATE_DB
+        createdSessions = self.get_exist_bfd_session() - bfdSessions
+        bfdSessions = self.get_exist_bfd_session()
+        assert len(createdSessions) == 1
+
+        session1 = createdSessions.pop()
+
+        # Check STATE_DB entry related to the BFD session
+        self.check_state_bfd_session_value(key1, fieldValues)
+
+        # Create BFD session 2
+        key2 = "default:default:10.0.1.2"
+        fieldValues = {"local_addr": "10.0.0.1", "tx_interval": "300", "rx_interval": "500"}
+        self.create_bfd_session(key2, fieldValues)
+        self.sdb.wait_for_n_keys(SOFT_BFD_STATE_TABLE, len(bfdSessions) + 1)
+
+        # Check BFD session 2 in STATE_DB
+        createdSessions = self.get_exist_bfd_session() - bfdSessions
+        bfdSessions = self.get_exist_bfd_session()
+        assert len(createdSessions) == 1
+
+        session2 = createdSessions.pop()
+
+        # Check STATE_DB entry related to the BFD session
+        self.check_state_bfd_session_value(key2, fieldValues)
+
+        # Create BFD session 3
+        key3 = "default:default:2000::2"
+        fieldValues = {"local_addr": "2000::1", "type": "demand_active"}
+        self.create_bfd_session(key3, fieldValues)
+        self.sdb.wait_for_n_keys(SOFT_BFD_STATE_TABLE, len(bfdSessions) + 1)
+
+        # Check BFD session 3 in STATE_DB
+        createdSessions = self.get_exist_bfd_session() - bfdSessions
+        bfdSessions = self.get_exist_bfd_session()
```

Comment:
can you please add a test in test_vnet.py where a vnet route with BFD session is created and added, that way we can be sure that it works fine with VNET routes as well.

Reply:
@prsunny is software BFD on DPU over VNET a requirement? Currently not supported